src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -589,42 +606,76 @@ public void testCatchErrorInProcessFilesMethod() throws Exception {
 
     @Test
     public void testExternalConfigurationResourceDoesNotExist() throws Exception {
-        final Checker mockChecker = createMockCheckerWithCacheForModule(DummyFileSetCheck.class);
+        final File cacheFile = temporaryFolder.newFile();
+        final Checker mockChecker = createMockCheckerWithCacheForModule(
+            DummyFileSetCheck.class, cacheFile);
 
         final String pathToEmptyFile = temporaryFolder.newFile("EmptyFile.java").getPath();
         final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
 
         // We invoke 'verify' twice to invalidate cache
         // and have two identical exceptions which happen on the same line between runs
         final int numberOfRuns = 2;
+        final String[] md5AfterFirstAndSecondRuns = new String[numberOfRuns];
+        final Properties cache = new Properties();
         for (int i = 0; i < numberOfRuns; i++) {
             verify(mockChecker, pathToEmptyFile, expected);
```

Comment:
> if we change the line of verify 

Not just `verify` but anything that is in the stack trace too. So that won't work.

> I cannot figure out how to intersect the exception

I can't see anything either unless we use powermock to force a specific exception that we control.

> Do you mean that we should make sure that getting non_existing_external_resource.xml before the invocation of verify method in the loop throws an exception

That sounds like what we will have to do.